### ISSUE TITLE
Monitoring v2 - some deployment&daemonSet do not work in RKE2

### ISSUE THREAD

**Issue by jiaqiluo** [rancher/rke2#277]
Steps:
- run rancher: `v2.5.0-alpha1` in HA mode
- import an RKE2 cluster
- go to dashboard -> apps -> install monitoring v2

Results:
The monitoring app is installed successfully, and most parts of it are functional. 
There are only the following 2 issues found:

- The daemonSet `pushprox-kube-controller-manager-client` and `pushprox-kube-scheduler-client` cannot find the control plane node because no nodes in RKE2 have the following label

    ```node-role.kubernetes.io/controlplane = true```

 

- The pods of the daemonSet `pushprox-kube-etcd-client` cannot be active because of the following error

```
Warning | FailedMount | Unable to attach or mount volumes: unmounted volumes=[metrics-cert-dir-source], unattached volumes=[metrics-cert-dir-source metrics-cert-dir default-token-mzqkd]: timed out waiting for the condition | 3 minutes ago
Warning | FailedMount | MountVolume.SetUp failed for volume "metrics-cert-dir-source" : hostPath type check failed: /etc/kubernetes/ssl is not a directory | 9 minutes ago
Warning | FailedMount | Unable to attach or mount volumes: unmounted volumes=[metrics-cert-dir-source], unattached volumes=[metrics-cert-dir default-token-mzqkd metrics-cert-dir-source]: timed out waiting for the condition | 17 minutes ago
Warning | FailedMount | Unable to attach or mount volumes: unmounted volumes=[metrics-cert-dir-source], unattached volumes=[default-token-mzqkd metrics-cert-dir-source metrics-cert-dir]: timed out waiting for the condition | 19 minutes ago
Normal | Scheduled | Successfully assigned cattle-monitoring-system/pushprox-kube-etcd-client-ng9gx to ip-172-31-33-217 | Invalid date
```

![image](https://user-images.githubusercontent.com/6218999/91774443-a2168880-eb9d-11ea-8155-42f424bc7f23.png)

**Comment by davidnuzik** [comment]
@jiaqiluo what were the rke2 nodes running on ec2? I'm just wanting to verify they were not running on DigitalOcean as we had a somewhat related bug on that. Although I see nodes have internal IP so guessing not DO.

**Comment by jiaqiluo** [comment]
> @jiaqiluo what were the rke2 nodes running on ec2? I'm just wanting to verify they were not running on DigitalOcean as we had a somewhat related bug on that. Although I see nodes have internal IP so guessing not DO.

Yes, they are EC2 instances

**Comment by cjellick** [comment]
Yeah, no nodes in rke2 are going to have the `node-role.kubernetes.io/controlplane` label because that isn't an upstream label. They'll only have `node-role.kubernetes.io/master: "true"`.
So, I don't know why the pushprox daemonset is looking for that role for rke2

**Comment by cjellick** [comment]
@jiaqiluo any idea if this works for one of the other cluster types defined in our monitoring chart, like k3s or kubeadm?
@aiyengar2 As far as I can tell, the rke2 side of the chart rancher-monitoring chart is setting things properly to use the "master" label, but its not getting fed down into the pushprox chart

**Comment by jiaqiluo** [comment]
> @jiaqiluo any idea if this works for one of the other cluster types defined in our monitoring chart, like k3s or kubeadm?

I am planning to test monitoring on those clusters, will let you know how it works.

**Comment by aiyengar2** [comment]
@cjellick it does seem like we're properly using the master label on the chart based on 
https://github.com/rancher/charts/blob/6293280499af87bbe4905c15d8e938d1c273ac4e/packages/rancher-monitoring/rancher-monitoring.patch#L1701-L1714

@jiaqiluo if you are using the UI to test the RKE2 deployment rather than manually enabling the flags, the change to the UI has not been merged in yet and is being tracked here: https://github.com/rancher/dashboard/issues/1130. If you selected the provider to be RKE, you are likely using the RKE1 PushProx exporters instead, which will not work. This seems to match up with the label you mentioned based on https://github.com/rancher/charts/blob/6293280499af87bbe4905c15d8e938d1c273ac4e/packages/rancher-monitoring/rancher-monitoring.patch#L1523-L1535.

**Comment by cjellick** [comment]
I'm assigning this to @aiyengar2 as I think it is in his wheel house
